### DISCUSSION TITLE
Wish List

### DISCUSSION THREAD

**Discussion by luni64** [luni64/VisualTeensy#19] [Ideas]
## List with wishes / requirements for VisualTeensy

please add ideas...

**Comment by luni64** [comment]
1.  Copy **static assets** (e.g. .gitgnore, .clang-format etc) to the project folder. Source and target location shall be selectable.  see #18. This might go into the settings tab, not in the project tab. 

2. Add **configuration management**. Add a possibility to define different configurations (e.g. T3.5, T4.0, various optimizer settings etc). On build the configuration shall be selectable, additionally a "build all" option shall be available (building all configurations). This might be useful for developing libraries which should compile under various conditions. 

3. Add a **library development mode**. This should generate a library project compliant to https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification. Code should not be generated from a src/main.cpp but from one of the examples in examples/someExample/main.cpp. Ideally a couple of example stubs should be generated. Might be useful to use *.ino for the examples in this mode. 

4. Possibility to set the **color of the left side bar**. This comes in handy if you work on projects with more than one Teensy. 

5. Improve **Documentation**

6. Add more target boards. E.g. Teensy 2 (#12), ESP32, Uno..

**Comment by T-Py-T** [comment]


**Reply by luni64** [reply]
Maybe I misunderstand something but  I can not imagine that this works.  All the Teensyduino low level code directly writes / reads from fixed memory locations (look at kintetis.h or imxrt.h). How should that work when running on a PC?

**Reply by luni64** [reply]
There is also a software only interface to GDB. It has a few rough edges but it might be worth a try: 

https://forum.pjrc.com/threads/61373-Using-GDB-with-Teensy-without-hardware-debugger-first-Beta

**Reply by T-Py-T** [reply]
I "think" I am compiling to an executable that will run on windows, and the GDB is going through that. 
I am trying to catch up on how much I really don't know. Thank you for being patient

That link seems to be exactly what I am trying to do.

I can remove my main comment from this list. Just let me know

**Comment by luni64** [comment]
No, just leave it. Might be interesting for others as well. Let me know how your experience is with the debugger. When I saw the thread I was considering adding the setup to vsTeensy but I never found time for it.
